code/src/main/java/habit_mode/model/Habit.java

```diff
@@ -0,0 +1,103 @@
+package habit_mode.model;
+
+/** The habit class
+ * 
+ * @author	CS 3212
+ * @version Spring 2022
+ */
+public class Habit {
+	public static final String NULL_TEXT_ERROR = "text for the habit cannot be null";
+	public static final String EMPTY_TEXT_ERROR = "text for the habit cannot be empty";
+	private String text;
+	private boolean complete;
+	private Frequency completionFrequency;
+
+	/** Creates a new habit.
+	 * 
+	 * @precondition string != null && string != string.isEmpty()
+	 * @postcondition this.getText() = string, this.isComplete() = false, this.getFrequency() = frequency;
+	 * 
+	 * @param text The text to display for the habit.
+	 * @param frequency How frequently the habit should be completed.
+	 */
+	public Habit(String text, Frequency frequency) {
```

Comment:
precondition and postcondition use string, but param name is text. Not sure if this is a problem or not but thought I would point it out